rfcs/1111-first-class-virtual-device.md

```diff
@@ -0,0 +1,120 @@
+- Feature Name: Add virtual device information as a first class field on Relay expressions.
+- Start Date: 2021-11-20
+- RFC PR: [apache/tvm-rfcs#1111](https://github.com/apache/tvm-rfcs/pull/1111)
+- GitHub Issue: [apache/tvm#1111](https://github.com/apache/tvm/issues/1111)
+
+# Summary
+[summary]: #summary
+
+I propose adding a new field to Relay expressions, virtual_device_. This field will contain virtual device information [currently called SEScope].
+
+# Motivation
+[motivation]: #motivation
+
+ Currently, the virtual device information (called SEScope today, but we will rename it soon) is stored in on_device Relay ops. This op is a wrapper op that contains the virtual device information for an expression. This pattern method was used to encode device information in Relay before @mbs-octoml's recent work on device planning. In his work, he continued to use on_device to encode the virtual device information.
```

Comment:
"on_device" annotations and Function attributes

I'd not bother with the before/after dev planning stuff, just take main as given